helper/logging/logging_http_transport_test.go

```diff
@@ -0,0 +1,233 @@
+package logging_test
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"regexp"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-log/tflogtest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/logging"
+)
+
+func TestNewLoggingHTTPTransport(t *testing.T) {
+	ctx, loggerOutput := setupRootLogger()
+
+	transport := logging.NewLoggingHTTPTransport(http.DefaultTransport)
+	client := http.Client{
+		Transport: transport,
+		Timeout:   10 * time.Second,
+	}
+
+	reqBody := `An example
+		multiline
+		request body`
+	req, _ := http.NewRequest("GET", "https://www.terraform.io", bytes.NewBufferString(reqBody))
+	res, err := client.Do(req.WithContext(ctx))
+	if err != nil {
+		t.Fatalf("request failed: %v", err)
+	}
+	defer res.Body.Close()
+
+	entries, err := tflogtest.MultilineJSONDecode(loggerOutput)
+	if err != nil {
+		t.Fatalf("log outtput parsing failed: %v", err)
+	}
+
+	if len(entries) != 2 {
+		t.Fatalf("unexpected amount of logs produced; expected 2, got %d", len(entries))
+	}
+
+	reqEntry := entries[0]
+	if diff := cmp.Diff(reqEntry, map[string]interface{}{
+		"@level":              "debug",
+		"@message":            "An example multiline request body",
+		"@module":             "provider",
+		"tf_http_op_type":     "request",
+		"tf_http_req_method":  "GET",
+		"tf_http_req_uri":     "/",
+		"tf_http_req_version": "HTTP/1.1",
+		"Accept-Encoding":     "gzip",
+		"Host":                "www.terraform.io",
+		"User-Agent":          "Go-http-client/1.1",
+		"Content-Length":      "37",
+	}); diff != "" {
+		t.Fatalf("unexpected difference for logging of the request:\n%s", diff)
+	}
+
+	resEntry := entries[1]
+	expectedResEntryFields := map[string]interface{}{
+		"@level":                    "debug",
+		"@module":                   "provider",
+		"Content-Type":              "text/html",
+		"tf_http_op_type":           "response",
+		"tf_http_res_status_code":   float64(200),
+		"tf_http_res_version":       "HTTP/2.0",
+		"tf_http_res_status_reason": "200 OK",
+	}
+	for ek, ev := range expectedResEntryFields {
+		if resEntry[ek] != ev {
+			t.Fatalf("Unexpected value for field %q; expected %q, got %q", ek, ev, resEntry[ek])
+		}
+	}
+
+	expectedNonEmptyEntryFields := []string{
+		"@message", "Etag", "Date", "X-Frame-Options", "Server",
```

Comment:
Similar note about potentially using a new `tf_http_res_body` field and setting message to "Received HTTP response" so its clear what folks are seeing in the logs.

Reply:
As above.
I can go this way, I don't mind. Just wanted a final check about the log masking limitations we would encounter.

Reply:
Done.